src/loongarch64/context.rs

```diff
@@ -153,10 +171,49 @@ impl TaskContext {
                 crate::asm::flush_tlb(None); // currently flush the entire TLB
             }
         }
+        #[cfg(feature = "fp_simd")]
+        unsafe {
+            save_fp_registers(&mut self.fp_status);
```

Comment:
Make `save_fp_registers` and `restore_fp_registers` member functions of `FpStatus` and call them in the following way:

```rs
self.fp_status.save();
next_ctx.fp_status.restore();
```